gaia/geo/gdal_functions.py

```diff
@@ -176,16 +196,45 @@ def world_to_pixel(geoMatrix, x, y):
         polygon_json = json.dumps(polygon_json)
     poly = ogr.CreateGeometryFromJson(polygon_json)
 
-    # Convert the layer extent to image pixel coordinates
+
     min_x, max_x, min_y, max_y = poly.GetEnvelope()
-    ul_x, ul_y = world_to_pixel(geo_trans, min_x, max_y)
-    lr_x, lr_y = world_to_pixel(geo_trans, max_x, min_y)
+    pixelPoly = world_to_pixel_poly(geo_trans, poly)
+
+    # Convert the layer extent to image pixel coordinates
+    ul_x, lr_x, ul_y, lr_y = pixelPoly.GetEnvelope()
```

Comment:
a more Pythonic way of combining these two lines is

    ul_x, lr_x, ul_y, lr_y = map(int, pixelPoly.GetEnvelope())